test/test_config.h.in

```diff
@@ -9,16 +9,33 @@
 #define RENDER_ENGINE_VALUES ::testing::ValuesIn(\
     gz::rendering::TestValues())
 
+#include <vector>
+#include <gz/common/Util.hh>
+#include <gz/rendering/config.hh>
+
 /// \todo(anyone) re-enable ogre2 test once ogre 2.2 works on macOS
 #ifdef __APPLE__
 static const std::vector<const char *> kRenderEngineTestValues{"ogre", "optix"};
 #else
-static const std::vector<const char *> kRenderEngineTestValues{"ogre2", "optix"};
+  #if defined(HAVE_OGRE) && defined(HAVE_OPTIX) && defined(HAVE_OGRE2)
+  static const std::vector<const char *> kRenderEngineTestValues{"ogre", "ogre2", "optix"};
+  #elif defined(HAVE_OGRE) && defined(HAVE_OPTIX)
+  static const std::vector<const char *> kRenderEngineTestValues{"ogre", "optix"};
+  #elif defined(HAVE_OGRE) && defined(HAVE_OGRE2)
+  static const std::vector<const char *> kRenderEngineTestValues{"ogre", "ogre2"};
```

Comment:
see above, we can't have `ogre` running with `ogre2`, so this would be:

```
static const std::vector<const char *> kRenderEngineTestValues{"ogre2"};
```

Reply:
[a712899](https://github.com/gazebosim/gz-rendering/pull/1089/commits/a712899aa114ac32d351ad1a51afe6a9e1327bff)